src/Common/src/Interop/Linux/System.Net.Security.Native/Interop.HeimdalNtlmException.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class NetSecurity
+    {
+        internal sealed class HeimdalNtlmException : Exception
+        {
+            public HeimdalNtlmException(string message) : base(message)
+            {
+            }
+
+            public HeimdalNtlmException(int error)
+                : base(SR.Format(SR.net_generic_heimntlm_operation_failed, error))
+            {
+                HResult = error;
```

Comment:
Is there any string-based description of the error we can retrieve from the library?  Or if these errors are standard errno values, can we use the existing infrastructure we have for getting exceptions/strings for such errors?